mungegithub/mungers/assign-reassign-handler.go

```diff
@@ -0,0 +1,170 @@
+func (AssignReassignHandler) RequiredFeatures() []string { return []string{} }
+
+// Initialize will initialize the munger
+func (AssignReassignHandler) Initialize(config *github.Config, features *features.Features) error {
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (AssignReassignHandler) EachLoop() error { return nil }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (h AssignReassignHandler) AddFlags(cmd *cobra.Command, config *github.Config) {
+}
+
+// Munge is the workhorse the will actually make updates to the PR
+func (h AssignReassignHandler) Munge(obj *github.MungeObject) {
+	if !obj.IsPR() {
+		return
+	}
+
+	comments, err := obj.ListComments()
+	if err != nil {
+		glog.Errorf("unexpected error getting comments: %v", err)
+		return
+	}
+
+	fileList, err := obj.ListFiles()
+	if err != nil {
+		glog.Errorf("Could not list the files for PR %v: %v", obj.Issue.Number, err)
+		return
+	}
+
+	//get all the people that could potentially own the file based on the blunderbuss.go implementation
+	potentialOwners, _ := getPotentialOwners(obj, h.features, fileList)
+
+	toAssign, toUnassign := h.assignOrRemove(obj, comments, fileList, potentialOwners)
+
+	//assign and unassign reviewers as necessary
+	for _, username := range toAssign.List() {
+		obj.AssignPR(username)
+	}
+	if len(toUnassign) > 0 {
+		is := goGithub.IssuesService{}
+		is.RemoveAssignees(repoOwner, *obj.Issue.Repository.Name, *obj.Issue.Number, toUnassign.List())
+	}
+}
+
+//assignOrRemove checks to see when someone comments "/assign" or "/reassign"
+// "/assign" self assigns the PR
+// "/reassign" unassignes the commenter and reassigns to someone else
+// [TODO] "/reassign <github handle>" reassign to this person
+func (h *AssignReassignHandler) assignOrRemove(obj *github.MungeObject, comments []*goGithub.IssueComment, fileList []*goGithub.CommitFile, potentialOwners weightMap) (toAssign, toUnassign sets.String) {
+	toAssign = sets.String{}
+	toUnassign = sets.String{}
+
+	for i := len(comments) - 1; i >= 0; i-- {
+		comment := comments[i]
+		if !mungerutil.IsValidUser(comment.User) {
+			continue
+		}
+
+		fields := getFields(*comment.Body)
+		if isAssignComment(fields) {
+			//check if they are a valid reviewer if so, assign the user. if not, explain why
+			if isValidReviewer(potentialOwners, comment.User) {
+				glog.Infof("Assigning %v to review PR#%v", *comment.User.Login, obj.Issue.Number)
+				toAssign.Insert(*comment.User.Login)
+			} else {
+				//inform user that they are not a valid reviewer
+				obj.WriteComment(fmt.Sprintf(notReviewerInTree, comment.User.String()))
+			}
+		}
+		if isReassignComment(fields) && isAssignee(obj.Issue.Assignees, comment.User) {
+			//check if they are already an assigned reviewer. if so, remove them.  if not, do nothing.
+			glog.Infof("Removing %v as an reviewer for PR#%v", *comment.User.Login, obj.Issue.Number)
+			toUnassign.Insert(*comment.User.Login)
```

Comment:
Same problem we had with `/approve`/`/approve cancel`. What if I do `/unassign` and then `/assign`? I will be in both `toAssign` and `toUnassign`.